src/main/java/org/klodnicki/repository/BookRepository.java

```diff
@@ -1,16 +1,31 @@
 package org.klodnicki.repository;
 
 import jakarta.persistence.EntityManager;
-import org.klodnicki.entity.Book;
+import jakarta.persistence.TypedQuery;
+import org.klodnicki.entity.BookInfo;
 
 public class BookRepository {
 
     private final EntityManager entityManager = HibernateUtil.getSessionFactory().createEntityManager();
 
-    public void add(Book book) {
+    public void add(BookInfo bookInfo) {
         entityManager.getTransaction().begin();
-        entityManager.persist(book);
+        entityManager.persist(bookInfo);
         entityManager.getTransaction().commit();
     }
 
+    public BookInfo findBookByTitleAndAuthor(String title, String author) {
```

Comment:
What about i.e. "edition"? Parameters "title" and "author" might be not specific enough in the future, as there will be let's say, multiple editions of the same title and author.

Reply:
as I reader I don't care which edition it is so I thought there were more people like me. If the library wants to have some choice by entering the edition please let me know, for now I skip it

Reply:
But reader doesn't provide those information. The library employee do. There will be a problem with which exactly book did they lent as there is no choice if more than 1 book was returned by search you implemented. 

P.S. Don't resolve conversations that are not resolved yet. If you say "done" and provide changes that's okay, but if you like here, provide more feedback to the issue, then please leave it unresolved :)